src/rtos/FreeRTOS.c

```diff
@@ -58,15 +57,15 @@ struct FreeRTOS_params {
 };
 
 struct FreeRTOS_thread_entry {
-	struct list_head list;
 	threadid_t threadid;
 	target_addr_t tcb;
 };
 
 struct FreeRTOS {
 	const struct FreeRTOS_params *param;
 	unsigned thread_count;
-	struct list_head thread_list;
+	/* Map from tcb to FreeRTOS_thread_entry. */
```

Comment:
So this is the motivation for pulling in the gnulib stuff, to get optimized performance on the lookups and maintenance of this thread_map?

Reply:
Basically. The real motivation is not the performance so much as not having to write the code myself. (There are actually lots of places within OpenOCD that could benefit from using well-implemented data structures, and assuming this gets accepted upstream I plan to use these structures more in the future.)

Reply:
Makes sense.  I guess if upstream community balks at gnulib for some reason, there's a remedy at that time, but leveraging gnulib more widely seems like a good direction to try to move toward.